src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -1320,17 +1321,23 @@ public MockUnsafeValues getUnsafe()
 	}
 
 	@Override
-	public BlockData createBlockData(Material material)
+	public @NotNull BlockData createBlockData(@NotNull Material material)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Validate.isTrue(material != null, "Must provide material");
+		return new BlockDataMock(material);
```

Comment:
This is fine for now but when we get to the point where we have multiple `BlockData` classes and mocks, a static `BlockDataMock.mock(material)` method would be more suitable, this way we can use a switch statement and instantiate the corresponding BlockData class.